pkg/daemon/update.go

```diff
@@ -2051,6 +2052,39 @@ func (dn *CoreOSDaemon) experimentalUpdateLayeredConfig() error {
 			if err := client.RebaseLayered(desiredImage); err != nil {
 				return err
 			}
+
+			state, err := client.GetStatusStructured()
+			if err != nil {
+				return err
+			}
+			for _, deployment := range state.Deployments {
+				if deployment.Staged {
+					staged = deployment
+				}
+			}
+
+			// since we reboot whenever kargs change, we can get old kargs from live-applied or booted
+			oldContentBytes, err := Cat(booted.Osname, booted.Checksum, render.MCDContentPath)
+			if err != nil {
+				return err
+			}
+			newContentBytes, err := Cat(staged.Osname, staged.Checksum, render.MCDContentPath)
+			if err != nil {
+				return err
+			}
+			var oldContent, newContent render.MCDContent
+			if err := json.Unmarshal(oldContentBytes, &oldContent); err != nil {
+				return err
+			}
+			if err := json.Unmarshal(newContentBytes, &newContent); err != nil {
+				return err
+			}
+			// this shouldn't break transactionality because if
+			// 1. power is lost: rpm-ostree won't have changed the bootloader entry yet, so we'll reboot into the old image, and the rebase will be run again
+			// 2. we can't have graceful shutdown, because the MCO installs a SIGTERM handler
+			if err := dn.updateKernelArguments(oldContent.KernelArguments, newContent.KernelArguments); err != nil {
```

Comment:
Need to think about a way to make this idempotent. I got into an interesting state where I'd failed to apply a karg (I think it was invalid?) but I couldn't apply the correct karg because --delete'ing the one that never got applied will fail